include/git2/mailmap.h

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (C) the libgit2 contributors. All rights reserved.
+ *
+ * This file is part of libgit2, distributed under the GNU GPL v2 with
+ * a Linking Exception. For full terms see the included COPYING file.
+ */
+#ifndef INCLUDE_git_mailmap_h__
+#define INCLUDE_git_mailmap_h__
+
+#include "common.h"
+#include "types.h"
+#include "buffer.h"
+
+/**
+ * @file git2/mailmap.h
+ * @brief Mailmap parsing routines
+ * @defgroup git_mailmap Git mailmap routines
+ * @ingroup Git
+ * @{
+ */
+GIT_BEGIN_DECL
+
+/**
+ * Allocate a new mailmap object.
+ *
+ * This object is empty, so you'll have to add a mailmap file before you can do
+ * anything with it. The mailmap must be freed with 'git_mailmap_free'.
+ *
+ * @param out pointer to store the new mailmap
+ * @return 0 on success, or an error code
+ */
+GIT_EXTERN(int) git_mailmap_new(git_mailmap **out);
+
+/**
+ * Free the mailmap and its associated memory.
+ *
+ * @param mm the mailmap to free
+ */
+GIT_EXTERN(void) git_mailmap_free(git_mailmap *mm);
+
+/**
+ * Add a single entry to the given mailmap object. If the entry already exists,
+ * it will be replaced with the new entry.
+ *
+ * @param mm mailmap to add the entry to
+ * @param real_name the real name to use, or NULL
+ * @param real_email the real email to use, or NULL
+ * @param replace_name the name to replace, or NULL
+ * @param replace_email the email to replace
+ * @return 0 if it was added, EEXISTS if it replaced an entry, or an error code
+ */
+GIT_EXTERN(int) git_mailmap_add_entry(
```

Comment:
I'm sorry for the last-minute review, but I don't love the `EEXISTS` return here.  That really makes it sound like an error case, not a success.  Do you think that "replaced an entry" is really such a valuable state to know?  If so, what do you think about an out-param that indicates that instead of overloading the return value?  Curious what @pks-t thinks.

Reply:
The reasoning behind this decision was to match the git_vector APIs here. Most consumers of the mailmap API will probably just get it from a git_repository which just ignores errors like this.

I'd probably be fine with just not communicating whether or not an entry was replaced entirely.

Reply:
Maybe an `int force` parameter (instead of a int `*replaced`) ?

Reply:
> I'd probably be fine with just not communicating whether or not an entry was replaced entirely.

I'm +1 on this.